### ISSUE TITLE
Make the tooltip follow the cursor in Piechart.

### ISSUE THREAD

**Issue by SAGARACH65** [recharts/recharts#1970]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
In pie charts making the tooltip follow the cursor like in the case with other charts such as radial charts will surely be useful. 

### What does the proposed API look like?
I thinks this can be a default implementation like such in the case of radial charts. If not it can be done by passing a prop to the `Tooltip` component

**Comment by xile611** [comment]
Thanks for your suggestion, there seems no other supporters. We may change the default behavior If it's a strong requirement.

**Comment by siberry** [comment]
I also need the tooltip to follow the cursor on the Piechart, like it does on other charts

**Comment by dougludlow** [comment]
I stumbled onto this issue because I want to be able to change the default tooltip functionality, I want the tooltip to follow the cursor.

**Comment by ckifer** [comment]
You should be able to hack this to make it work for now if anyone wants
https://codesandbox.io/p/sandbox/get-cell-renderer-forked-wxmdx7?workspaceId=5d3c7e6a-4f10-4d63-b56a-6e222ec3935f

**Comment by AryaTheAbhishek** [comment]
> You should be able to hack this to make it work for now if anyone wants https://codesandbox.io/p/sandbox/get-cell-renderer-forked-wxmdx7?workspaceId=5d3c7e6a-4f10-4d63-b56a-6e222ec3935f

It goes out of the container also not working for pie chart

**Comment by ckifer** [comment]
> It goes out of the container

Yes that was what was desired by this particular user. Once you're outside of recharts with workarounds like this it would be on you to keep it within the container if you want

> not working for pie chart

https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-345k9d
Seems to work just fine

**Comment by AryaTheAbhishek** [comment]
> > It goes out of the container
> 
> Yes that was what was desired by this particular user. Once you're outside of recharts with workarounds like this it would be on you to keep it within the container if you want
> 
> > not working for pie chart
> 
> https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-345k9d Seems to work just fine



Can you help me with keeping tooltip inside the viewbox providing custom position encountering issue where the tooltip overflows the viewbox when the chart area is limited.

https://github.com/user-attachments/assets/54c27758-efd5-4ac5-a996-1c3dbc8a6a30
https://github.com/user-attachments/assets/ab362dc9-fbdc-4ac4-8409-0f2b86b3aaa1

In the default tooltip implementation, it automatically adjusts and inverts its position (either to the left, right, top, or bottom) when there’s no enough space for it within the viewbox by keeping allowEscapeViewBox as false

https://github.com/user-attachments/assets/73e263bf-cd3d-497a-ba68-8c8e9adf1f00

**Comment by ckifer** [comment]
Here you are
https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-345k9d

You need to do the math for the positioning yourself

```jsx
          left: shouldFlipX
            ? // when we flip X subtract the tooltip width
              screenCoords.x - OFFSET - TT_WIDTH
            : screenCoords.x + OFFSET,
```

![Image](https://github.com/user-attachments/assets/a4003fba-bdc3-4d37-915c-eafaf3e9589a)
